src/main/kotlin/com/exactpro/th2/woodpecker/Service.kt

```diff
@@ -113,9 +118,11 @@ class Service(
             onInfo { "Started load cycle ${cycle + 1}" }
 
             forEach { step ->
+                onStart(step.settings.readSettings())
                 onInfo { "Started load step: ${step.toHuman()}" }
-                repeat(step.duration) { yield(step.rate) }
+                repeat(step.duration * minBatchesPerSecond) { yield(step.rate / minBatchesPerSecond) }
```

Comment:
Is it correct to divide the rate by minBatchesPerSecond?
rate - messages per second 
minBatchesPerSecond - batch per second

Reply:
Yep, the more often we generate batches the less their size must be to sum to requested load value